runelite-client/src/main/java/net/runelite/client/plugins/birdhouses/BirdhousesOverlay.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2018, Kristian <https://github.com/kristian-welsh>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.birdhouses;
+
+import com.google.inject.Inject;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.util.Arrays;
+import java.util.List;
+import net.runelite.api.Client;
+import net.runelite.api.TileObject;
+import net.runelite.api.VarPlayer;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+class BirdhousesOverlay extends Overlay
+{
+	private static final List<Integer> EMPTY_STATES = Arrays.asList(
+		1,
+		4,
+		7,
+		10,
+		13,
+		16,
+		19,
+		22,
+		25
+	);
+	
+	private final BirdhousesPlugin plugin;
+	
+	@javax.inject.Inject
+	private Client client;
+	
+	@Inject
+	public BirdhousesOverlay(BirdhousesPlugin plugin)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		this.plugin = plugin;
+	}
+	
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!plugin.isNearBirdhouses())

```

Comment:
If `plugin.getBirdhouses()` is empty whenever a player is outside of the birdhouse regions, this check won't be needed since the TileObject loop below won't be executed. This seems unnecessary.